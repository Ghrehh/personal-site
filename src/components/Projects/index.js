import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.module.css';

class Projects extends React.Component {
  renderProjects() {
    return this.props.projects.map((project, index) => (
      <section key={index}>
        <h2 className={styles.name}>{project.name}</h2>
        <p>{project.description}</p>
      </section>
    ));
  }

  render() {
    return (
      <article className={styles.projects}>{this.renderProjects()}</article>
    );
  }
}

Projects.propTypes = {
  projects: PropTypes.array.isRequired
}

export default Projects;
