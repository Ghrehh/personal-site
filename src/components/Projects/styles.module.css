.header {
  font-size: 40px;
  font-weight: 800;
  text-align: center;
  padding: 45px 0 40px;
}

.project {
  text-align: center;
  cursor: pointer;
  padding: 40px 15px;
  text-align: center;
  box-shadow: 1px 1px 5px 0px rgba(0,0,0,0.28);
  max-width: 450px;
  margin: auto;
  margin-bottom: 20px;
  border-radius: 3px;
  transition: transform 0.5s, filter 0.5s;
}

.project:hover {
  transform: translate(0, -4px);
  filter: brightness(1.1);
}

.name {
  margin: auto;
  max-width: 400px;
  color: white;
  letter-spacing: 3px;
  font-size: 55px;
  margin-bottom: 15px;
  font-family: var(--header-font);
  font-size: 30px;
}

.description {
  margin: auto;
  max-width: 300px;
  color: white;
  font-size: 14px;
}

@media (max-width: 530px) {
  .project {
    max-width: 100%;
    margin: 0;
    box-shadow: none;
    border-radius: 0;
  }

  .project:hover {
    transform: initial;
  }
}
