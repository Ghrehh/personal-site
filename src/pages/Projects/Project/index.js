// @flow
import React from 'react';

import * as projectModel from 'models/data/project';
import * as pathConstants from 'constants/paths';
import SectionHeading from 'components/text/SectionHeading';
import UnstyledLink from 'components/UnstyledLink';

import styles from './styles.module.css';

type Props = {
  project: projectModel.Type,
  path: string
};

const Project = (props: Props) => (
  <div className={styles.projectOuter}>
    <UnstyledLink link={`${pathConstants.PROJECTS}/${props.path}`}>
      <div
        style={{ backgroundColor: props.project.backgroundColor }}
        className={styles.project}
      >
        <div className={styles.projectInner}>
          <SectionHeading className={styles.projectName}>
            {props.project.name}
          </SectionHeading>

          <p className={styles.projectBody}>{props.project.description}</p>
        </div>
      </div>
    </UnstyledLink>
  </div>
);

export default Project;
