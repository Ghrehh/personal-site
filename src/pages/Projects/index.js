import React from 'react';
import PropTypes from 'prop-types';

import Page from 'components/Page';

import styles from './styles.module.css';

class Projects extends React.Component {
  renderProjects() {
    return this.props.projects.map((project, index) => (
      <section
        className={styles.project}
        key={index}
        style={{ backgroundColor: project.backgroundColor }}
      >
        <h2 className={styles.name}>{project.name}</h2>
        <p className={styles.description}>{project.description}</p>
      </section>
    ));
  }

  render() {
    return (
      <Page>
        <article>
          <header>
            <h2 className={styles.header}>PROJECTS</h2>
          </header>

          {this.renderProjects()}
        </article>
      </Page>
    );
  }
}

Projects.propTypes = {
  projects: PropTypes.array.isRequired
};

export default Projects;
