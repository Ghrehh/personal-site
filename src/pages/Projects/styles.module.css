.title {
  color: var(--red);
  margin-bottom: 55px;
}

.buttonContainer {
  text-align: center;
  margin-bottom: 40px;
}
