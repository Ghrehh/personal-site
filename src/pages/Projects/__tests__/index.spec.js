import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Projects from '../';

describe('Projects', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};

    const projects = [
      { name: 'foo', description: 'bar', backgroundColor: 'red' }
    ];

    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(<Projects projects={projects} />);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
