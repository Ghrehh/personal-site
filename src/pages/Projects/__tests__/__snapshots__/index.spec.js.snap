// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects renders the component correctly 1`] = `
<FadeIn
  timeout={0}
>
  <Page>
    <header
      className="header"
    >
      <PageHeading
        className="title"
      >
        PROJECTS
      </PageHeading>
      <p
        className="headerBody"
      >
        See below for a selection of my open-source projects. Click on a project for more information.
      </p>
    </header>
    <div
      className="buttonContainer"
    >
      <LinkButton
        link="/"
      >
        Back
      </LinkButton>
    </div>
    <Project
      path="nsts-metronome"
      project={
        Object {
          "backgroundColor": "var(--green)",
          "description": "A sleek web metronome capable of playing in non-standard time signatures. Built using Vue JS",
          "gitHubLink": "https://github.com/Ghrehh/nsts-metronome",
          "name": "NSTS METRONOME",
          "tags": Array [
            "Vue",
            "Music",
            "CSS animation",
          ],
          "websiteLink": "https://nsts-metronome.github.io/",
        }
      }
    />
    <Project
      path="ear-training"
      project={
        Object {
          "backgroundColor": "var(--red)",
          "description": "A web app for musicians to train their aural perception skills.",
          "gitHubLink": "https://github.com/Ghrehh/ear-training",
          "name": "EAR TRAINING",
          "tags": Array [
            "Music",
          ],
          "websiteLink": "https://ghrehh.github.io/eartraining/",
        }
      }
    />
    <Project
      path="vue-world-map"
      project={
        Object {
          "backgroundColor": "var(--light-blue)",
          "description": "A Vue.js library for displaying heat map style data per country on a world map.",
          "gitHubLink": "https://github.com/Ghrehh/vue-world-map",
          "name": "VUE WORLD MAP",
          "tags": Array [
            "Package",
          ],
        }
      }
    />
    <Project
      path="tweetatlas-client"
      project={
        Object {
          "backgroundColor": "var(--yellow)",
          "description": "Client used to display data from the Tweetatlas Server, built using Vue JS.",
          "gitHubLink": "https://github.com/Ghrehh/tweetatlas-client",
          "name": "TWEETATLAS CLIENT",
          "tags": Array [
            "Vue",
            "WebSockets",
          ],
          "websiteLink": "https://tweetatlas.github.io/",
        }
      }
    />
    <Project
      path="tweetatlas-server"
      project={
        Object {
          "backgroundColor": "var(--purple)",
          "description": "Service used to estimate the location of users creating tweets containing a search phrase or phrases. Built using Go.",
          "gitHubLink": "https://github.com/Ghrehh/tweetatlas-server",
          "name": "TWEETATLAS SERVER",
          "tags": Array [
            "Go",
            "WebSockets",
          ],
        }
      }
    />
    <Project
      path="portfolio-website"
      project={
        Object {
          "backgroundColor": "var(--green)",
          "description": "The website you're currently viewing. Built using Create React App.",
          "gitHubLink": "https://github.com/Ghrehh/portfolio-site",
          "name": "PORTFOLIO WEBSITE",
          "tags": Array [
            "React",
            "Flow",
          ],
        }
      }
    />
  </Page>
</FadeIn>
`;
